metrics-otlp/README.md

```diff
@@ -0,0 +1,159 @@
+     -f secret-values.yaml
+   ```
+
+1. Verify the deployment.
+   Check that the related Pod has been created in the Namespace and is in the `Running` state:
+   ```bash
+   kubectl -n $KYMA_NS rollout status deploy metrics-gateway
+   ```
+
+## Install the agent
+
+1. Deploy the agent.
+
+   The gateway deployed in the previous step can be used to _push_ metrics via OTLP. If you prefer to use the Prometheus _pull_ approach, you need a dedicated instance to scrape your workload. Furthermore, to retrieve node-specific metrics, you need an agent as a DaemonSet.
+   
+   Deploy an Otel Collector using the upstream Helm chart with a prepared [values](./metrics-agent-values.yaml). The values file defines a metrics pipeline for scraping workload by annotation and pushing them to the gateway. It also defines a second metrics pipeline determining node-specific metrics for your workload from the nodes kubelet and the nodes filesystem itself.
+
+   ```bash
+   helm upgrade metrics-agent open-telemetry/opentelemetry-collector --version 0.47.0 --install --namespace $KYMA_NS \
+     -f https://raw.githubusercontent.com/kyma-project/examples/main/metrics-otlp/metrics-agent-values.yaml \
+     --set config.exporters.otlp.endpoint=metrics-gateway.$KYMA_NS:4317
+   ```
+
+1. Verify the deployment.
+   Check that the related Pod has been created in the Namespace and is in the `Running` state:
+   ```bash
+   kubectl -n $KYMA_NS rollout status daemonset metrics-agent
+   ```
+
+## Result
+
+Now, you have a setup in place as outlined in the [architecture diagram](#architecture), with a gateway and an agent. By default, the node-specific metrics are automatically ingested by the agent, and pushed via the gateway to the configured backend. Furthermore, metrics for the gateway and agent instances are exported.
+
+
+## Usage
+
+To add metrics ingestion for your custom workload, you have the following options:
+
+### Prometheus pull-based, with annotations
+
+This approach assumes that you instrumented your application using a library like the [prometheus client library](https://prometheus.io/docs/instrumenting/clientlibs/), having a port in your workload exposed serving a typical Prometheus metrics endpoint.
+
+The agent is configured with a generic scrape configuration, which uses annotations to specify the endpoints to scrape in the cluster. 
+Having the annotations in place is everything you need for metrics ingestion to start automatically.
+
+Put the following annotations either to a service that resolves your metrics port, or directly to the pod:
+
+```yaml
+prometheus.io/scrape: "true"   # mandatory to enable automatic scraping
+prometheus.io/scheme: https    # optional, default is "http" if no Istio sidecar is used. When using a sidecar (Pod has label `security.istio.io/tlsMode=istio`), the default is "https". Use "https" to scrape workloads using Istio client certificates.
```

Comment:
The pod-endpoints scrape_config does not have a tls_config. Was that intentional? Then we should give a hint that https works only on services.

Reply:
Very good finding, have not tested the final setup once more.